pkg/scannerv4/mappers/mappers.go

```diff
@@ -633,22 +674,40 @@ type severityValues struct {
 	v2Score  float32
 	v3Vector string
 	v3Score  float32
+	source   v4.VulnerabilityReport_Vulnerability_CVSS_Source
+	url      string
 }
 
-// severityAndScores returns the severity and scores information out of a
-// ClairCore vulnerability. The returned information is dependent on the
-// underlying updater.
-func severityAndScores(ctx context.Context, vuln *claircore.Vulnerability, nvdVuln *nvdschema.CVEAPIJSON20CVEItem) (severityValues, error) {
+// severityAndCvssMetrics processes the severity and CVSS scores
+// for a given vulnerability based on the updater type (RHEL, NVD, OSV, etc.).
+// The returned data includes CVSS metrics, ACS preferred severity values, and potential errors
+// encountered during the calculation process for each source.
+func severityAndCvssMetrics(ctx context.Context, vuln *claircore.Vulnerability, nvdVuln *nvdschema.CVEAPIJSON20CVEItem) ([]*v4.VulnerabilityReport_Vulnerability_CVSS, severityValues, error) {
 	ctx = zlog.ContextWithValues(ctx,
-		"component", "mappers/severityAndScores",
+		"component", "mappers/severityAndCvssMetrics",
 		"updater", vuln.Updater,
 		"vuln_id", vuln.ID,
 		"vuln_name", vuln.Name,
 	)
-
+	var cvssMetrics []*v4.VulnerabilityReport_Vulnerability_CVSS
```

Comment:
hmm I've been trying to see if there is a way to do this cleaner, but I'm having trouble... Perhaps this is the way to go

Reply:
I have updated logics of `toCVSSMetrics()` and `severityAndCvssMetrics()`
Let me know what you think